cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowToken.java

```diff
@@ -20,9 +20,47 @@
 import javax.annotation.Nullable;
 
 /**
- * Interface to represent the data that is transferred from one node to the next node in the {@link Workflow}
+ * Interface to represent the data that is transferred from one node to the next node in the {@link Workflow}.
  */
 public interface WorkflowToken {
+
+  /**
+   * Put the specified key-value entry in the {@link WorkflowToken}.
+   * The value is stored against the node at which it is being added.
+   * @param key   the key representing the entry
+   * @param value the value for the key
+   */
+  void putValue(String key, String value);
```

Comment:
I would call this setValue, because below you have a method getLastSetter(). Or call that one getLastPutter()? (that sounds strange)